src/shmem_team.c

```diff
@@ -287,6 +287,7 @@ int shmem_internal_team_split_strided(shmem_internal_team_t *parent_team, int PE
         if (myteam->psync_idx == -1 || myteam->psync_idx >= shmem_internal_params.TEAMS_MAX) {
             RAISE_WARN_MSG("No more teams available (max = %ld), try increasing SHMEM_TEAMS_MAX\n",
                             shmem_internal_params.TEAMS_MAX);
+            /* No psync is available, but must call barrier across parent team before returning. */
```

Comment:
Not sure I understand the comment. Is this to cleanup after the above reduction?

Reply:
It may not be necessary.  It's really just a reminder that we cannot return here because all PEs need to finish the barrier across the parent team below.  My intent was also to clarify the check for `myteam->psync_idx == -1` _after_ this parent team barrier.  Maybe a cross-reference between the 2 comments would make that more clear?  Or maybe we just don't need the comments...

Reply:
Got it. I think a comment on the barrier below might be helpful. That barrier is there to eliminate a race with back-to-back team creations, right?